### PR TITLE
Client report graphs

### PR DESCRIPTION
**Pull request by thomassth**
- Fixed beeswarm graph
- Added participants graph

### CODE REVIEW

**Comment by patcon** [comment]
Thanks @thomassth! Still reviewing, but first thought was that there's already some test data and helper functions in .storybook/utils.js that other stories are using. Could we stick with using those functions, and add more if any from your testData file are missing?

See: https://github.com/CivicTechTO/polis-storybook/blob/main/.storybook/utils.js
and 
https://github.com/CivicTechTO/polis-storybook/blob/3e5000f46ebeeefdc48b8869ba258a647ca12e75/stories/compdem/client-participation/TidCarousel.stories.js#L6-L10

Also, I spawned #19, because added more data sources is a really nice idea, and lets us actually start using storybook to look at past snapshots of data using the visualizations hosted here (which is its own killer feature)

**Comment by patcon** [comment]
Maybe in scope of this PR, you could version control your new data files alongside the originals? (for now, if we could stick with the current sample data, as I'm pretty sure some stories would break if other data was swapped in, and generalizing them seems out-of-scope for this issue. But if you added the data to the repo, viewing it for now would be as simple as changing the conversation_id in the imports when working locally)

https://github.com/CivicTechTO/polis-storybook/blob/3e5000f46ebeeefdc48b8869ba258a647ca12e75/.storybook/utils.js#L2-L3

So versioning in your data into these files:

```
.storybook/data/5esrbenwxs-participation-init.json
.storybook/data/5esrbenwxs-comments.json
```

(participation init endpoint is a composite of enough data to init the visualization on first load, and is at: https://pol.is/api/v3/participationInit?conversation_id=5esrbenwxs )

**Comment by patcon** [comment]
Lemme know what you think! Thanks again 🙌

**Comment by patcon** [comment]
And unless there's a reason to change the submodule, would love to keep it pointed to the "storybook-prep" branch, which is a bare minimal changeset from upstream, just to get storybook working

If we're diverging too much, then we'll just add a new submodule in `codebases/civictechto` and create alternative stories for our components that differ from upstream. Right now, the `stories/compdem/*` is for things building on their base codebase

**Comment by thomassth** [comment]
I've looked through the `storybook-prep` branch, I believe many of the changes should be merged too! It's beneficial to keep the code style updated & prep for future, bigger package upgrades.

**Comment by thomassth** [comment]
I have removed the excess test data for now; we can think about adding them in other commits.

**Comment by patcon** [comment]
Hm, I'm not getting it building properly in storybook on local -- same for CI:
https://civictechto.github.io/polis-storybook/PR-18/

Found the build error line (it doesn't fail out, and maybe that's something to improve in the build process):
https://github.com/CivicTechTO/polis-storybook/actions/runs/11433350904/job/31805190108?pr=18#step:3:1025

Did you have it working locally? Anything that maybe didn't get committed?

I'll update if I sort it out myself :) Thanks @thomassth 🙌 🙌

**Comment by patcon** [comment]
K, gonna merge in mainline, where I've added a fix for ensuring failures show in CI when they happen in the build :)
#22

**Comment by patcon** [comment]
K, failure confirmed, and now I think I've got a fix. Needed to add some more packages to the storybook environment, and make sure they get added as a d3 global

**Comment by patcon** [comment]
k yay!

ParticipantsGraph working now 🎉  https://civictechto.github.io/polis-storybook/PR-18/?path=/story/compdem-client-report-participantsgraph--default

~~Beeswarm still needs some work:~~ We got beeswarm working too!
https://civictechto.github.io/polis-storybook/PR-18/?path=/story/compdem-client-report-beeswarm--default

Now that you've got us going in the right direction, I think I know the fixes for this :)

File: .storybook/utils.js [hunk 1]
```diff
@@ -17,14 +20,21 @@ export const getExtremity = () => {
...
 }
 
+// Simulates response data from /api/v3/comments?conversation_id=3ntrtcehas&moderation=true&include_voting_patterns=true
 export const getComments = () => {
   return commentsData
 }
 
+// Simulates response data from /api/v3/reports?report_id=r3bpnywujybyru4rkx92i
+export const getReports = () => {
+  return reportsData
```
Comment (patcon) [inline .storybook/utils.js:35]: Aligned this with the response from actual API endpoint (an array)

File: stories/compdem/client-report/ParticipantsGraph.stories.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  repfulAgreeTidsByGroup: null,
+  math: null,
+  renderHeading: null,
+  report: null,
+  voteColors: null,
+}
+
+const comments = getComments();
+
+// Logic from client-report/src/components/app.js#L274-291
```
Comment (patcon) [inline stories/compdem/client-report/ParticipantsGraph.stories.js:16]: <3 THANK YOU SO MUCH. This was what I was missing 🙏  🙏  🙏  🙏

**Comment by thomassth** [comment]
Ah that's why, I didn't know the storybook needed another round of imports

**Comment by patcon** [comment]
~~Hm, it's working locally, but I'm seeing errors in the deployed storybook -- trying to figure out what's up with that~~ 🤔  

~~Maybe related: https://github.com/storybookjs/storybook/discussions/21525~~

EDIT: Ok, feeling good that I've pushed the fix 🤞

**Comment by patcon** [comment]
Ok, they're both working in the deployed storybook. Merge away if it looks good to you!

Thanks for getting this in!

**Review by patcon — APPROVED** [review]

**Pull request merged**